addons/account/models/account_move.py

```diff
@@ -2733,16 +2731,15 @@ def _post(self, soft=True):
 
         # Create the analytic lines in batch is faster as it leads to less cache invalidation.
         to_post.mapped('line_ids').create_analytic_lines()
-        to_post.write({
-            'state': 'posted',
-            'posted_before': True,
-        })
 
         for move in to_post:
             move.message_subscribe([p.id for p in [move.partner_id] if p not in move.sudo().message_partner_ids])
 
             # Compute 'ref' for 'out_invoice'.
             if move._auto_compute_invoice_reference():
+                # We need the name of the move so that it can be used as a reference for
+                # the invoice line before the move is hashed.
+                move.with_context(force_compute_name=True)._compute_name()
```

Comment:
Because posting the moves will create the hash. But then, these statements https://github.com/odoo/odoo/pull/112320/files#diff-1e3bd6be3bfb83a37ec9fb800ce8b1c95afe0be90ff792874ae7299c320a2f6eR2748-R2749
will try to write on the name of the lines which is not permitted now because we also use the name of the lines in the hash.
So that's why I have to force the computation of a name before the move is posted

Reply:
Wouldn't it make sense to include all of this in a function
```python
@api.depends('move_type', 'name')
def _compute_payment_reference():
```
And by adding the `move_id.name` in the dependences of `<account.move.line>._compute_name`, at least in master (and maybe 16.0)?
I really don't like the `force_compute_name`